src/ledger/LedgerManagerImpl.cpp

```diff
@@ -293,104 +294,117 @@ LedgerManagerImpl::loadLastKnownLedger(function<void()> handler)
+            LedgerHeader lh;
+            CLOG_INFO(Ledger,
+                      "Setting empty ledger while core rebuilds state: {}",
+                      ledgerAbbrev(lh));
+            setLedgerTxnHeader(lh, mApp);
+            latestLedgerHeader = lh;
+        }
+        else
+        {
+            auto currentLedger =
+                LedgerHeaderUtils::loadByHash(getDatabase(), lastLedgerHash);
+            if (!currentLedger)
             {
-                LedgerHeader lh;
-                CLOG_INFO(Ledger,
-                          "Setting empty ledger while core rebuilds state: {}",
-                          ledgerAbbrev(lh));
-                setLedgerTxnHeader(lh, mApp);
+                throw std::runtime_error("Could not load ledger from database");
             }
-            else
+            HistoryArchiveState has = getLastClosedLedgerHAS();
+            if (currentLedger->ledgerSeq != has.currentLedger)
             {
-                auto currentLedger = LedgerHeaderUtils::loadByHash(
-                    getDatabase(), lastLedgerHash);
-                if (!currentLedger)
-                {
-                    throw std::runtime_error(
-                        "Could not load ledger from database");
-                }
-                HistoryArchiveState has = getLastClosedLedgerHAS();
-                if (currentLedger->ledgerSeq != has.currentLedger)
-                {
-                    throw std::runtime_error(
-                        "Invalid database state: last known "
-                        "ledger does not agree with HAS");
-                }
-
-                CLOG_INFO(Ledger, "Loaded LCL header from database: {}",
-                          ledgerAbbrev(*currentLedger));
-                setLedgerTxnHeader(*currentLedger, mApp);
+                throw std::runtime_error("Invalid database state: last known "
+                                         "ledger does not agree with HAS");
             }
+
+            CLOG_INFO(Ledger, "Loaded LCL header from database: {}",
+                      ledgerAbbrev(*currentLedger));
+            setLedgerTxnHeader(*currentLedger, mApp);
+            latestLedgerHeader = *currentLedger;
         }
-        else
+    }
+    else
+    {
+        // In no-history mode, this method should only be called when
+        // the LCL is genesis.
+        releaseAssertOrThrow(mLastClosedLedger.hash == lastLedgerHash);
+        releaseAssertOrThrow(mLastClosedLedger.header.ledgerSeq ==
+                             GENESIS_LEDGER_SEQ);
+        CLOG_INFO(Ledger, "LCL is genesis: {}",
+                  ledgerAbbrev(mLastClosedLedger));
+        latestLedgerHeader = mLastClosedLedger.header;
+    }
+
+    releaseAssert(latestLedgerHeader.has_value());
+
+    // Step 3. Restore BucketList if we're doing a full core startup
+    // (startServices=true), OR when using BucketListDB
+    if (startServices || mApp.getConfig().EXPERIMENTAL_BUCKETLIST_DB)
```

Comment:
We should never check the raw flag, used `getConfig().isUsingBucketListDB()` instead.